armi/reactor/reactors.py

```diff
@@ -127,12 +128,16 @@ def add(self, container):
             if len(cores) != 1:
                 raise ValueError(
                     "Only 1 core may be specified at this time. Please adjust input. "
-                    "Cores found: {}".format(cores)
+                    f"{len(cores)} cores found."
                 )
             self.core = cores[0]
 
-        if isinstance(container, SpentFuelPool):
-            self.sfp = container
+        if isinstance(container, ExcoreStructure):
+            nomen = container.name.replace(" ", "").lower()
+            if nomen == "spentfuelpool":
+                # special case
+                nomen = "sfp"
```

Comment:
I can add that, sure. But this will go away when I do the ticket: https://github.com/terrapower/armi/issues/1871

This only exists to backwards support a couple in-house TP projects.

Reply:
This will go away when ARMI no longer forces a default SFP on people.